file/pkg/service/service.go

```diff
@@ -15,14 +15,77 @@
 package service
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/micro/go-micro/v2/client"
+	"github.com/opensds/multi-cloud/file/pkg/db"
 	pb "github.com/opensds/multi-cloud/file/proto"
 	log "github.com/sirupsen/logrus"
 )
 
-type fileService struct {}
+type fileService struct {
+	fileClient pb.FileService
+}
 
 func NewFileService() pb.FileHandler {
 
 	log.Infof("Init file service finished.\n")
-	return &fileService{}
+	return &fileService{
+		fileClient: pb.NewFileService("file", client.DefaultClient),
+	}
+}
+
+func (f *fileService) ListFileShare(ctx context.Context, in *pb.ListFileShareRequest, out *pb.ListFileShareResponse) error {
+	log.Info("Received ListFileShare request.")
+
+	if in.Limit < 0 || in.Offset < 0 {
```

Comment:
Better to do this kind of paramater checking in api level.

Reply:
OK Sure.
Code Changes Done
Thanks